### PR TITLE
Python bindings

### PR DESCRIPTION
**Pull request by betaper**
I propose some minor updates related to the `python-bindings` code.
1. Exclude `.o` files from git (`.gitignore`)
2. Add some missing build prerequisites for `invoke` in `tasks.py`

### CODE REVIEW

File: python-bindings/tasks.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def build_cmult(c, path=None):
             c.run(path)
     else:
         print_banner("Building C Library")
-        cmd = "gcc -c -Wall -Werror -fpic cmult.c -I /usr/include/python3.7"
```
Comment (betaper) [inline python-bindings/tasks.py:60]: Not needed

File: python-bindings/tasks.py [hunk 2]
```diff
@@ -140,9 +140,9 @@ def compile_python_module(cpp_name, extension_name):
     invoke.run(
         "g++ -O3 -Wall -Werror -shared -std=c++11 -fPIC "
         "`python3 -m pybind11 --includes` "
-        "-I /usr/include/python3.7 -I .  "
```
Comment (betaper) [inline python-bindings/tasks.py:143]: The python include dir is already contained in the previous `pybind11` command

File: python-bindings/tasks.py [hunk 3]
```diff
@@ -140,9 +140,9 @@ def compile_python_module(cpp_name, extension_name):
...
+        "-I . "
         "{0} "
-        "-o {1}`python3.7-config --extension-suffix` "
```
Comment (betaper) [inline python-bindings/tasks.py:145]: More generic

File: python-bindings/tasks.py (hunk 1 above)
Comment (betaper) [inline python-bindings/tasks.py:60]: The python iclude dir is not needed here

**Review by jima80525 — APPROVED** [review]
Looks great!  Thanks for taking the time to post a PR on this.  All of these changes worked for me.

**Review by jima80525 — APPROVED** [review]
Looks like you'll have a rebase to main before this can get merged.

**Pull request merged**